Added position handler for appraisal window

oskomonzon opened the pull request:
This is to address #471 and hopefully cater a wide range of users usability concerns.

- [x] Added position handler to input dialog in appraisal mode
- [x] Added position handler logic
- [x] Implemented save, load and visibility logic for the position handler

Tagging @Blaisorblade, @nahojjjen, @DJCrapsody, @thearaks, @sarav and @Eremiell as usual for feedback and code review.

Blaisorblade commented:
Code LGTM (based on in-browser review). And this should not be very device-specific (though of course you never know), so this should be mergeable as soon as somebody else tests it.

nahojjjen commented:
Testing it, it seems to be working flawlessly 👍 

To be honest, with how well its working, I wouldn't be against having the window movement listener active even when appraisal is not active, on all of the overlay types.

nahojjjen commented:
Minor comments that might be worth considering:

1. Since its a button, the user might think the button moves the window up/down on click, and not understand that you have to drag it. Might be worth to Toast "drag to move window" if the user moves the window less than x pixels after the listener is done.

2. The padding is slightly off compared to the rest of the elements in the overlay.

oskomonzon commented:
I'm glad you guys like it. I've been thinking about @nahojjjen first point and it makes sense, adding a toast upon release if there was no position change.
Or maybe it isn't a terrible idea to have it behave like a toggle between "bottom" and "custom position" if the user just taps on it.

Blaisorblade commented:
What's the status of this PR? There's the status feedback from Travis to fix, and the comments from @nahojjjen to address. @oskomonzon are you waiting for feedback or just low on time?

oskomonzon commented:
@Blaisorblade very low of time, I've been out of the city again during the weekend.
I'll add the toast as sugested by @nahojjjen unless any of you guys have another proposal to discuss.

oskomonzon commented:
Rebased to keep it clean. This are the last changes:

- [ ] Merged and cleaned conflicts with master
- [ ] Added toast to instruct the user to move the window if he just tapped the handler
- [ ] Added new string to the strings resource files for the new toast
- [ ] Small changes to make it Travis compliant

Blaisorblade commented on app/src/main/java/com/kamron/pogoiv/Pokefly.java:
```diff
@@ -968,7 +963,7 @@ private void moveOverlay(Boolean moveUp) {
         if (moveUp) {
             newParams.gravity = Gravity.TOP;
-            newParams.y = sharedPref.getInt(APPRAISAL_WINDOW_POSITION, 0);
+            newParams.y = sharedPref.getInt(APPRAISAL_WINDOW_POSITION,0);
```
that's not good ;-)

oskomonzon replied:
Lol, I was so happy it passed the test.
